### PR TITLE
OVN controller implementation for Windows

### PR DESCRIPTION
**Pull request by JocelynBerrendonner**
Full, untested, implementation of the OVN controller code for Windows.

On Init, each Windows node creates a virtual switch and prepares the switch for overlay networking
Every time a new node is added to the K8s cluster, each other node creates a VXLAN route towards the newly added node
Whenever a node is removed from the K8s cluster, each other node removes the VXLAN route towards the node that was removed.
Tests: pending

### CODE REVIEW

File: go-controller/extensions/pkg/node/node_windows.go [hunk 1]
```diff
@@ -14,18 +15,49 @@ import (
 	"github.com/Microsoft/hcsshim/hcn"
 )
 
+const (
+
+	// Annotations consumed by this module (i.e. this module expects the following annotations to be already set)
+	// extensionDRMac is also set by InitSelf() on this node
+	extensionHostSubnet = "hybrid-sdn.extensions.ovn.kubernetes.io/hostsubnet" // Host subnet
+	extensionDRMac      = "hybrid-sdn.extensions.ovn.kubernetes.io/drmac"      // Distributed router/gateway MAC
```
Comment (mkostersitz) [inline go-controller/extensions/pkg/node/node_windows.go:23]: Nitpick: Should this not be extensionDrMac?

File: go-controller/extensions/pkg/node/node_windows.go [hunk 2]
```diff
@@ -14,18 +15,49 @@ import (
...
+	extensionDRIp       = "hybrid-sdn.extensions.ovn.kubernetes.io/drip"       // Distributed router/gateway IP address
```
Comment (mkostersitz) [inline go-controller/extensions/pkg/node/node_windows.go:24]: Nitpick: Should this not be extensionDrIp?

File: go-controller/extensions/pkg/node/node_windows.go [hunk 3]
```diff
@@ -40,12 +72,32 @@ func (n *nodeController) Add(node *kapi.Node) {
...
+
+	if !ok {
+		log.Errorf("Can't retreive the host subnet from the '%s' node's annotations", node.Name)
+		return
+	}
+
+	nodeIPAddress := getFirstInternalIp(node.Status.Addresses)
+
+	if nodeIPAddress == nil {
+		log.Errorf("Can't retreive node '%s' provider address from the node's annotations", node.Name)
```
Comment (mkostersitz) [inline go-controller/extensions/pkg/node/node_windows.go:85]: I would reword this to "Can't retreive node '%s' IP address from the node's annotations" to better reflect the step where we are failing since we are looking for the node's IP address not the provider address

**Review by mkostersitz — COMMENTED** [review]
Looks good as far as I can tell. A few nit picks and in general I would review it again for casing of variables and function names to make sure it is correct. 
Should we add a comment at the beginning that this is Microsoft Code and have a disclaimer in there?

File: go-controller/extensions/pkg/node/node_windows.go (hunk 1 above)
Comment (JocelynBerrendonner) [reply go-controller/extensions/pkg/node/node_windows.go:23]: This is a good question! I am not sure if there is a convention for these kind of naming, so I was using the convention I am familiar with. :) Any convention works with me!

File: go-controller/extensions/pkg/node/node_windows.go (hunk 3 above)
Comment (JocelynBerrendonner) [reply go-controller/extensions/pkg/node/node_windows.go:85]: That's fair. Let me improve the comment.

**Comment by JocelynBerrendonner** [comment]
@mkostersitz, I am abandoning this PR to rebase against @dcbw repository. I will address your feedback over there.
